### PR TITLE
Move YGLogger into YGConfig and associate YGNodeRef with log events

### PR DESCRIPTION
**Pull request by woehrl01**
Moves the `YGLogger` into `YGConfig` and pass the `YGNodeRef` into the logger to be able to associate the log messages and assertions with the specific node.

Tackles facebook/yoga#530 and facebook/yoga#446

### CODE REVIEW

File: yoga/YGMacros.h [hunk 1]
```diff
@@ -44,8 +44,7 @@
 #ifndef YG_ASSERT
 #define YG_ASSERT(X, message)              \
   if (!(X)) {                              \
-    YGLog(YGLogLevelError, "%s", message); \
-    YG_ABORT();                            \
+    YGLog(NULL, YGLogLevelError, "%s", message); \
```
Comment (emilsjolander) [inline yoga/YGMacros.h]: YGLogLevelFatal?

Comment (emilsjolander) [inline yoga/YGMacros.h]: YG_ASSERT does not seem to be used anyore, remove it?

File: yoga/Yoga.h [hunk 2]
```diff
@@ -219,7 +219,8 @@ YG_NODE_LAYOUT_EDGE_PROPERTY(float, Border);
 YG_NODE_LAYOUT_EDGE_PROPERTY(float, Padding);
 
 WIN_EXPORT void YGSetLogger(YGLogger logger);
-WIN_EXPORT void YGLog(YGLogLevel level, const char *message, ...);
+WIN_EXPORT void YGLog(const YGNodeRef node, YGLogLevel level, const char *message, ...);
+WIN_EXPORT void YGLogWithConfig(const YGConfigRef config, YGLogLevel level, const char *format, ...);
```
Comment (emilsjolander) [inline yoga/Yoga.h]: let's skip this until we feel the need for it. jsut use YGLog and pass a NULL node. Thoughts?

File: yoga/YGMacros.h (hunk 1 above)
Comment (woehrl01) [inline yoga/YGMacros.h]: It's still used in `YGNodeList.c` will adapt and remove.

File: yoga/Yoga.h (hunk 2 above)
Comment (woehrl01) [inline yoga/Yoga.h]: `YGLogWithConfig` is required for the places were we don't have node, just a config, e.g. `YGNodeNewWithConfig` and `YGConfigSetPointScaleFactor``

Comment (emilsjolander) [inline yoga/Yoga.h]: Yeah what I mean is that let's just pass NULL as the node in those places. Passing the config isn't super useful as configs are usually statically allocated and don't have info in them like a node does.

Comment (woehrl01) [inline yoga/Yoga.h]: But we could at least declare this only inside `Yoga.c` ?

Comment (woehrl01) [inline yoga/Yoga.h]: Oh, we actually pass `NULL`, `YGConfigRef` is only used to call the correct logger 👍

Comment (emilsjolander) [inline yoga/Yoga.h]: Actually no let's keep it. I was confusing it with the actually logger function. don't mind me :p You are right 👍

**Pull request closed without merging**